sdcm/cluster.py

```diff
@@ -3191,12 +3175,31 @@ def check_cluster_health(self):
             self.log.debug('Cluster health check disabled')
             return
 
-        for node in self.nodes:
-            node.check_node_health()
+        nemesis_count = len(self.nemesis)
+
+        # Don't run health check in case parallel nemesis.
+        # TODO: find how to recognize, that nemesis on the node is running
+        if nemesis_count == 1:
+            for node in self.nodes:
+                node.check_node_health()
+
+        self.check_nodes_running_nemesis_count(nemesis_count)
 
         ClusterHealthValidatorEvent(type='done', name='ClusterHealthCheck', status=Severity.NORMAL,
                                     message='Cluster health check finished')
 
+    def check_nodes_running_nemesis_count(self, nemesis_count):
+        nodes_running_nemesis = [node for node in self.nodes if node.running_nemesis]
+
+        # Support parallel nemesis: nemesis_count is
+        if len(nodes_running_nemesis) <= nemesis_count:
+            return
+
+        message = "; ".join(f"{node.ip_address} ({'seed' if node.is_seed else 'non-seed'}): {node.running_nemesis}"
+                            for node in nodes_running_nemesis)
+        ClusterHealthValidatorEvent(type='warning', name='NodesNemesis', status=Severity.WARNING,
+                                    message=f"There are more then expected nodes running nemesis: {message}")
+
```

Comment:
I don't understand what are you trying to do here...

Reply:
I check how many nodes are running nemesis. If it's more then amount of nemesises - raise WARNING

Reply:
If I understand it correctly, it's not good enough.
You shouldn't check it in runtime but check the configuration (i.e. params.get)

What will happen if one of the nemesis threads dies? the clusterHealthValidator will kicks in?
What happens if there will be several stages of the test?

Reply:
Get nemesises amount from test parameter. Changed